tutorials/tour/_posts/2017-02-13-inner-classes.md

```diff
@@ -31,50 +31,32 @@ class Graph {
   }
 }
 ```
- 
-In our program, graphs are represented by a list of nodes. Nodes are objects of inner class `Node`. Each node has a list of neighbours, which get stored in the list `connectedNodes`. Now we can set up a graph with some nodes and connect the nodes incrementally:
- 
-```tut
-object GraphTest extends App {
-  val g = new Graph
-  val n1 = g.newNode
-  val n2 = g.newNode
-  val n3 = g.newNode
-  n1.connectTo(n2)
-  n3.connectTo(n1)
-}
-```
- 
-We now enrich the above example with types to state explicitly what the type of the various defined entities is:
- 
+This program represents a graph as a list of nodes (`var nodes`). Each node knows which other nodes it's connected to (`connectedNodes`). The `class Node` is a _path-dependent type_ because it is nested in the `class Graph`. Therefore, everything in the `connectedNodes` must be created using the class attached to Graph.
```

Comment:
"Each node knows which..." seems odd. Nodes don't know anything since they don't have thoughts. This would be like saying a chair knows that it has four legs.